src/system/time/services/rtc.cpp

```diff
@@ -7,8 +10,21 @@
  * @return false On failure
  */
 bool TimeProviderRTC::init() {
-    this->datetime = { 0, 0, 0, 0, 0, 0 };
-    return true;
+    if (! rtc.begin()) {
+        Serial.println("Couldn't find RTC");
+        Serial.flush();
+        return false;
+    }
+    else{
+        this->datetime = { 0, 0, 0, 0, 0, 0 };
+        DateTime now = rtc.now();
+        datetime.year   = now.year();
+        datetime.month  = now.month();
+        datetime.day    = now.day();
+        datetime.minute = now.minute();
+        datetime.second = now.second();
+        return true;
```

Comment:
This code should be on the `update` function, so every time you want to update the time, you call the rtc.